src/dataproxy/worker/SharedWorkerProvider.tsx

```diff
@@ -0,0 +1,102 @@
+import * as Comlink from 'comlink'
+import React, { useState, useEffect, ReactNode } from 'react'
+
+import { useClient } from 'cozy-client'
+import Minilog from 'cozy-minilog'
+
+import {
+  DataProxyWorker,
+  DataProxyWorkerContext,
+  DataProxyWorkerPartialState
+} from '@/dataproxy/common/DataProxyInterface'
+import { TabCountSync } from '@/dataproxy/common/TabCountSync'
+
+const log = Minilog('👷‍♂️ [SharedWorkerProvider]')
+
+export const SharedWorkerContext = React.createContext<
+  DataProxyWorkerContext | undefined
+>(undefined)
+
+interface SharedWorkerProviderProps {
+  children: ReactNode
+}
+
+let boardcastChannel: BroadcastChannel
+if (typeof BroadcastChannel === 'function') {
+  boardcastChannel = new BroadcastChannel('DATA_PROXY_BROADCAST_CHANANEL')
+}
+const tcs = new TabCountSync()
+window.addEventListener('unload', () => tcs.close())
+
+export const SharedWorkerProvider = React.memo(
+  ({ children }: SharedWorkerProviderProps) => {
+    const [worker, setWorker] = useState<DataProxyWorker | undefined>()
+    const [workerState, setWorkerState] = useState<DataProxyWorkerPartialState>(
+      {
+        status: 'not initialized',
+        tabCount: 0
+      }
+    )
+    const [tabCount, setTabCount] = useState(0)
+    const client = useClient()
+
+    useEffect(() => {
+      if (!client) return
+
+      const doAsync = async (): Promise<void> => {
+        log.log('Init SharedWorker')
```

Comment:
What is a `log.log` compared to `log.debug` ? :thinking:

Reply:
No difference, they are mapped to the same method. I just replace `console` with `log`.

I'll replace this

Reply:
[Fixed](https://github.com/cozy/cozy-web-data-proxy/compare/706f11d012a29e5971ca7e3b9ca740d220b21e05..d4c89625494b97d98a279dca53386d11c881e736)